src/test/scala/utils/EmbeddedKafka.scala

```diff
@@ -11,15 +11,22 @@ import uk.sky.fs2.kafka.topicloader.TopicLoader.topicPartitionOrder
 
 trait EmbeddedKafka[F[_]] {
 
-  def embeddedKafka(implicit kafkaConfig: EmbeddedKafkaConfig, F: Async[F]): Resource[F, KafkaServer] =
+  def embeddedKafkaConfigF(implicit F: Sync[F]): F[EmbeddedKafkaConfig] = for {
+    kafkaPort     <- RandomPort[F]
+    zooKeeperPort <- RandomPort[F]
+  } yield EmbeddedKafkaConfig(kafkaPort, zooKeeperPort, customBrokerProperties = Map("log.roll.ms" -> "10"))
+
+  def embeddedKafkaR(implicit kafkaConfig: EmbeddedKafkaConfig, F: Async[F]): Resource[F, KafkaServer] =
     Resource.make(F.delay(Underlying.start().broker))(server => F.delay(server.shutdown()).void)
 
   def createCustomTopic(topic: String, partitions: Int, topicConfig: Map[String, String])(implicit
       kafkaConfig: EmbeddedKafkaConfig,
       F: Async[F]
   ): F[NonEmptyList[TopicPartition]] =
     for {
-      _ <- F.fromTry(Underlying.createCustomTopic(topic = topic, topicConfig = topicConfig, partitions = partitions))
+      maybeCreate <-
+        F.delay(Underlying.createCustomTopic(topic = topic, topicConfig = topicConfig, partitions = partitions))
+      _           <- F.fromTry(maybeCreate)
```

Comment:
This is the main change. Previously we were wrapping `createCustomTopic`, a Try in `F.fromTry` which wasn't deferring the execution.